setup.py

```diff
@@ -1,17 +1,15 @@
 #!/usr/bin/env python
 import os
-
-from distutils.core import setup
 from distutils.command.build_py import build_py as _build_py
 from distutils.command.install_data import install_data as _install_data
+from distutils.core import setup
```

Comment:
`distutils` is deprecated and will be removed in Python 3.12: https://docs.python.org/3/library/distutils.html
`setup` should be imported from `setuptools`.

I also find all the extra things done in this `setup.py` quite hacky...